src/Select.js

```diff
@@ -952,7 +955,13 @@ const Select = React.createClass({
 	},
 
 	renderMenu (options, valueArray, focusedOption) {
-		if (options && options.length) {
+		if (this.props.multi && this.props.maxSelectedOptions && this.props.maxSelectedOptionsText && this.props.maxSelectedOptions <= this.getValueArray(this.props.value).length) {
```

Comment:
Why also check if `maxSelectedOptionsText` is truthy? If you don't want to show it… and there's a default value anyway.